app/gui2/src/components/ComponentBrowser.vue

```diff
@@ -34,10 +35,13 @@ onMounted(() => {
 
 const transform = computed(() => {
   const nav = props.navigator
-  const pos = props.position
-  return `${nav.transform} translate(${pos.x}px, ${pos.y}px) scale(${
-    1 / nav.scale
-  }) translateY(-100%)`
+  const translate = nav.translate
+  const position = {
+    x: Math.round((translate.x + props.position.x) * nav.scale),
+    y: Math.round((translate.y + props.position.y) * nav.scale),
+  }
+
```

Comment:
Vectors have math methods, so this can be written without splitting the coordinates.

```suggestion
  const position = nav.translate.add(props.position).scale(nav.scale)
```